libraries/AP_SafeRTL/AP_SafeRTL.h

```diff
@@ -0,0 +1,152 @@
+#pragma once
+
+#include <AP_Buffer/AP_Buffer.h>
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Common/Bitmask.h>
+#include <AP_Math/AP_Math.h>
+#include <AP_AHRS/AP_AHRS.h>
+#include <DataFlash/DataFlash.h>
+#include <GCS_MAVLink/GCS.h>
+
+// definitions and macros
+#define SAFERTL_ACCURACY_DEFAULT        2.0f    // default _ACCURACY parameter value.  Points will be no closer than this distance together.
+#define SAFERTL_POINTS_DEFAULT          150     // default _POINTS parameter value.  High numbers improve path pruning but use more memory and CPU for cleanup. Memory used will be 20bytes * this number.
+#define SAFERTL_POINTS_MAX              500     // the absolute maximum number of points this library can support.
```

Comment:
The slowest algorithm is `detect_loops`, which runs in O(n^2) time. On a pixhawk, I think there's no way that it could store a path longer than a couple hundred points. I never did any testing with extremely large paths on a good cpu, but my hunch is that CPU load would be too great to allow a path length anywhere close to 16,000.